libtransmission/rpc-server.cc

```diff
@@ -643,6 +643,57 @@ void rpc_server_start_retry_cancel(tr_rpc_server* server)
     server->start_retry_counter = 0;
 }
 
+#ifdef _WIN32
+int evhttp_bind_socket_win(struct evhttp* httpd, char* address, ev_uint16_t port)
+{
+    WORD wVersionRequested;
+    WSADATA wsaData;
+    wVersionRequested = MAKEWORD(2, 2);
+    int iResult = WSAStartup(wVersionRequested, &wsaData);
+    if (iResult != 0)
+    {
+        tr_logAddInfo(fmt::format(FMT_STRING("WSAStartup failed")));
+        return -1;
+    }
+    struct addrinfo* result = NULL;
+    struct addrinfo hints;
+    ZeroMemory(&hints, sizeof(hints));
+    hints.ai_family = AF_UNSPEC;
+    hints.ai_socktype = SOCK_STREAM;
+    hints.ai_protocol = IPPROTO_TCP;
+    hints.ai_flags = AI_PASSIVE;
+
+    int ret = getaddrinfo(address, std::to_string(port).c_str(), &hints, &result);
+    if (ret != 0)
+    {
+        tr_logAddInfo(fmt::format(FMT_STRING("getaddrinfo failed")));
```

Comment:
This message isn't exactly useful to the user, and isn't exactly suitable for info level. Also, aren't you supposed to fallback to `evhttp_bind_socket()` if `evhttp_bind_socket_win()` fails? I think I'd rather not log anything in this case.